### PR TITLE
Upload refactor

### PR DESCRIPTION
**Pull request by billbitt**
- removed socket.io
- updated client side upload process to use form data submission to spee.ch's publish api
- consolidated client-side publish and validation functions into objects so they would be out of the global scope
- updated the File record creation on the publish API to include the channel name in the File record if applicable
- updated the error handling code on the back end to better return the specific error to the client

### CODE REVIEW

File: config/speechConfig_example.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
```
Comment (kauffj) [inline config/speechConfig_example.js]: A more common convention for this is probably `speechConfig.example.js` or `speechConfig.js.example`.

File: README.md [hunk 2]
```diff
@@ -36,17 +33,17 @@ spee.ch is a single-serving site that reads and publishes images and videos to a
 
 #### POST
 * /api/publish
-  * example: `curl -X POST -F 'name=MyPictureName' -F 'nsfw=false' -F 'file=@/path/to/my/picture.jpeg' https://spee.ch/api/publish`
```
Comment (kauffj) [inline README.md]: @lyoshenka I gather best practices re: API documentation have evolved. Should @billbitt consider swagger or something else to document these calls?

(May be out of scope for this PR and should instead be opened separately)

File: auth/authentication.js [hunk 3]
```diff
@@ -2,12 +2,14 @@ const db = require('../models');
 const logger = require('winston');
 
 module.exports = {
-  authenticateChannelCredentials (channelName, userPassword) {
+  authenticateChannelCredentials (skipAuth, channelName, userPassword) {
```
Comment (kauffj) [inline auth/authentication.js]: Why call this then if `skipAuth` is true?

File: controllers/publishController.js [hunk 4]
```diff
@@ -6,24 +6,33 @@ const publishHelpers = require('../helpers/publishHelpers.js');
...
+          return db.Channel.findOne({where: {channelName: publishParams.channel_name}});
+        } else {
+          logger.debug('this claim was published in channel: n/a');
+          return null;
+        }
       })
       .then(channel => {
-        let certificateId;
+        // set channel information
+        certificateId = null;
```
Comment (kauffj) [inline controllers/publishController.js]: Missing `let`?

File: helpers/errorHandlers.js [hunk 5]
```diff
@@ -2,38 +2,60 @@ const logger = require('winston');
...
-      return error;
+      status = 400;
+      message = error;
     }
+    return [status, message];
+  },
+  handleRequestError: function (action, originalUrl, ip, error, res) {
+    logger.error(`Request Error on ${originalUrl}`, module.exports.useObjectPropertiesIfNoKeys(error));
+    postToStats(action, originalUrl, ip, null, null, error);
+    const errorStatusAndMessage = this.returnErrorMessageAndStatus(error);
```
Comment (kauffj) [inline helpers/errorHandlers.js]: [error, message] = this.returnErrorMessageAndStatus(error)

Prob worth spending some time reading/learning on ES2016 destructuring as well as spread operators.

File: helpers/errorHandlers.js [hunk 6]
```diff
@@ -2,38 +2,60 @@ const logger = require('winston');
...
+        .status(errorStatusAndMessage[0])
+        .render('requestError', {
+          status : errorStatusAndMessage[0],
+          message: errorStatusAndMessage[1],
+        });
+  },
+  handleApiError: function (action, originalUrl, ip, error, res) {
+    logger.error(`Api ${action} Error on ${originalUrl}`, module.exports.useObjectPropertiesIfNoKeys(error));
+    postToStats(action, originalUrl, ip, null, null, error);
+    const errorStatusAndMessage = this.returnErrorMessageAndStatus(error);
```
Comment (kauffj) [inline helpers/errorHandlers.js]: same as above

File: public/assets/js/publishFileFunctions.js [hunk 7]
```diff
@@ -1,99 +1,231 @@
-/* publish functions */
+var stagedFiles = null;
 
-function cancelPublish () {
-    window.location.href = '/';
-}
+var publishFileFunctions = {
```
Comment (kauffj) [inline public/assets/js/publishFileFunctions.js]: Presumably `const`

File: public/assets/js/publishFileFunctions.js [hunk 8]
```diff
@@ -1,99 +1,231 @@
...
+    triggerFileChooser: function (fileInputId) {
+        document.getElementById(fileInputId).click();
+    },
+    cancelPublish: function () {
```
Comment (kauffj) [inline public/assets/js/publishFileFunctions.js]: Late on catching this one, but this function name doesn't imply it is redirecting me to the homepage.

File: auth/authentication.js (hunk 3 above)
Comment (billbitt) [reply auth/authentication.js]: good point.  I was struggling with how to best do this in the promise chain, but I now abstracted this to its own function

File: config/speechConfig_example.js (hunk 1 above)
Comment (billbitt) [reply config/speechConfig_example.js]: I like `speechConfig.js.example`

File: controllers/publishController.js (hunk 4 above)
Comment (billbitt) [reply controllers/publishController.js]: defined on line 9

File: README.md (hunk 2 above)
Comment (billbitt) [reply README.md]: I'll discuss with @lyoshenka and open separate PR for needed changes

**Pull request merged**